### PR TITLE
Doc: use a glossary

### PR DESCRIPTION
**Pull request by emillon**
This switches our handmade *Terminology* section to a Sphinx glossary:

<https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#glossary>

The glossary itself is formatted in a better way, and it makes it possible to
link to definitions using the special syntax:

    :term:`library`
    :term:`libraries <library>`

Some examples are added. We probably do not want to link all instances of a
term, but this looks like it can be useful at the beginning of a section.

### CODE REVIEW

File: doc/overview.rst [hunk 1]
```diff
@@ -49,73 +49,84 @@ use the ``dune`` command.
...
+
+   build context root
+     The root of a build context named ``foo`` is ``<root>/_build/<foo>``.
+
+   build target
+     Specified on the command line, e.g., ``dune build <target_path.exe>``. All
+     targets that Dune knows how to build live in the ``_build`` directory.
+
+   alias
+     A build target that doesn't produce any file and has configurable
```
Comment (Alizter) [inline doc/overview.rst:107]: Should we mention that it helps to think of aliases as sets?

File: doc/overview.rst [hunk 2]
```diff
@@ -49,73 +49,84 @@ use the ``dune`` command.
...
+     dependencies. Targets starting with ``@`` on the command line are
```
Comment (Alizter) [inline doc/overview.rst:108]: Do we also mention `@@foo`?

File: doc/overview.rst [hunk 3]
```diff
@@ -49,73 +49,84 @@ use the ``dune`` command.
...
+     also trigger alias construction in all children directories recursively.
+     If no target is specified, Dune builds the ``default`` alias. Dune defines
+     several :ref:`builtin-aliases`.
+
+   environment
+     Determines the default values of various parameters, such as the
+     compilation flags. In Dune, each directory has an environment attached to
+     it. Inside a scope, each directory inherits the environment from its
+     parent. At the root of every scope, a default environment is used. At any
+     point, the environment can be altered using an :ref:`dune-env` stanza.
```
Comment (Alizter) [inline doc/overview.rst:120]: the stanza

File: doc/usage.rst [hunk 4]
```diff
@@ -179,10 +179,10 @@ targets upon starting:
...
 -------
 
-Targets starting with a ``@`` are interpreted as aliases. For instance
-``@src/runtest`` means the alias ``runtest`` in all descendants of
-``src`` in all build contexts where it is defined. If you want to
-refer to a target starting with a ``@``, simply write: ``./@foo``.
+Targets starting with a ``@`` are interpreted as :term:`aliases <alias>`. For
+instance ``@src/runtest`` means the alias ``runtest`` in all descendants of
+``src`` in all build contexts where it is defined. If you want to refer to a
+target starting with a ``@``, simply write: ``./@foo``.
```
Comment (Alizter) [inline doc/usage.rst:185]: Also mention `@@foo`?

**Review by Alizter — APPROVED** [review]

**Comment by emillon** [comment]
Thanks - I haven't touched the content so I'll leave it as is for now.

**Pull request merged**